src/dsp_tools/commands/xmlupload/transform_resource_and_values_for_upload.py

```diff
@@ -242,167 +225,257 @@ def _make_bitstream_file_value(bitstream_info: BitstreamInfo) -> dict[str, Any]:
+                g += _make_richtext_value_graph(
+                    val=val,
+                    prop_name=prop_name,
+                    res_bn=res_bn,
+                    permissions_lookup=permissions_lookup,
+                    iri_resolver=iri_resolver,
+                )
             case _:
                 assert_never(val.value)
-    return SerialiseProperty(property_name=prop.name, values=values)
+    return g
 
 
-def _transform_into_richtext_value(
-    val: XMLValue, permissions_lookup: dict[str, Permissions], iri_resolver: IriResolver
-) -> SerialiseRichtext:
+def _make_richtext_value_graph(
+    val: XMLValue,
+    prop_name: URIRef,
+    res_bn: BNode,
+    permissions_lookup: dict[str, Permissions],
+    iri_resolver: IriResolver,
+) -> Graph:
+    val_bn = BNode()
+    resolved_permission = _resolve_permission(val.permissions, permissions_lookup)
+    g = _add_optional_triples(val_bn, resolved_permission, val.comment)
     xml_val = cast(FormattedTextValue, val.value)
+    val_str = _get_richtext_string(xml_val, iri_resolver)
+    g.add((res_bn, prop_name, val_bn))
+    g.add((val_bn, RDF.type, KNORA_API.TextValue))
+    g.add((val_bn, KNORA_API.textValueAsXml, Literal(val_str, datatype=XSD.string)))
+    g.add((val_bn, KNORA_API.textValueHasMapping, URIRef("http://rdfh.ch/standoff/mappings/StandardMapping")))
+    return g
+
+
+def _get_richtext_string(xml_val: FormattedTextValue, iri_resolver: IriResolver) -> str:
     xml_with_iris = xml_val.with_iris(iri_resolver)
-    val_str = xml_with_iris.as_xml()
-    permission_str = _get_permission_str(val.permissions, permissions_lookup)
-    return SerialiseRichtext(value=val_str, permissions=permission_str, comment=val.comment)
+    return xml_with_iris.as_xml()
 
 
-def _transform_into_serialise_prop(
-    prop: XMLProperty,
-    permissions_lookup: dict[str, Permissions],
-    transformer: TransformationSteps,
-) -> SerialiseProperty:
-    serialised_values = [_transform_into_serialise_value(v, permissions_lookup, transformer) for v in prop.values]
-    prop_serialise = SerialiseProperty(
-        property_name=prop.name,
-        values=serialised_values,
-    )
-    return prop_serialise
+def _add_optional_triples(val_bn: BNode, permissions: str | None, comment: str | None) -> Graph:
+    g = Graph()
+    if permissions:
+        g.add((val_bn, KNORA_API.hasPermissions, Literal(permissions, datatype=XSD.string)))
+    if comment:
+        g.add((val_bn, KNORA_API.valueHasComment, Literal(comment, datatype=XSD.string)))
+    return g
 
 
-def _transform_into_serialise_value(
-    value: XMLValue,
-    permissions_lookup: dict[str, Permissions],
-    transformer: TransformationSteps,
-) -> SerialiseValue:
-    permission_str = _get_permission_str(value.permissions, permissions_lookup)
-    value_str = cast(str, value.value)
-    return transformer.serialiser(value_str, permission_str, value.comment)
+def _add_optional_permission_triple(
```

Comment:
I want to keep that because we can use it once we also serialise the other file values with rdflib (files don't have comments). I will do that in a separate PR

Reply:
It is not dead, it is just only used once, so it felt like overkill to have a separate function to me.